src/riscv/lib/src/jit/builder/control_flow_graph.rs

```diff
@@ -136,6 +136,30 @@ impl<'info, T> BudgetCheckLoc<'info, T> {
     }
 }
 
+/// Information about a step budget check.
+///
+/// A budget is the maximum number of steps that can be taken from this point
+/// without exceeding the overall step budget for the sequence.
+#[derive(Debug)]
+pub struct BudgetCheck<'info, T> {
+    /// Budget associated with this budget check.
+    budget: usize,
+    /// Edge where the budget check occurs.
+    edge: &'info DirectedEdgeInfo<T>,
+}
+
+impl<'info, T> BudgetCheck<'info, T> {
+    /// Return the budget associated with this budget check.
```

Comment:
mega nit
```suggestion
    /// Returns the budget associated with this budget check.
```
(and below for `edge`)